e2e/stress/DeviceTelemetryTest/DeviceTelemetryTest.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Client;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    public class DeviceTelemetryTest
+    {
+        int _messageSize;
+        int _messageCount;
+        int _parallelCount;
+        TransportType _protocol;
+        ResultWriter _writer;
+        DeviceClient _client;
+        Stopwatch _wall = new Stopwatch();
+
+        public DeviceTelemetryTest(
+            TransportType protocol,
+            int messageCount,
+            int messageSize,
+            int parallelCount,
+            ResultWriter resultWriter)
+        {
+            _messageCount = messageCount;
+            _messageSize = messageSize;
+            _protocol = protocol;
+            _parallelCount = parallelCount;
+            _writer = resultWriter;
+
+            if (_messageCount < _parallelCount) throw new ArgumentException($"{nameof(messageCount)} needs to be greater than {nameof(parallelCount)}");
+        }
+
+        public void CreateDevice()
+        {
+            string connectionString =
+                Environment.ExpandEnvironmentVariables(Environment.GetEnvironmentVariable("IOTHUB_DEVICE_CONN_STRING"));
+
+            _client = DeviceClient.CreateFromConnectionString(connectionString, _protocol);
+
+            _wall.Start();
+        }
+
+        public async Task SendMessageAsync()
+        {
+            var metrics = new DeviceTelemetryMetrics();
+            metrics.MessageSize = _messageSize;
+            metrics.Protocol = _protocol.ToString();
+            metrics.WallTime = _wall.Elapsed.TotalMilliseconds;
+
+            try
+            {
+                Stopwatch watch = new Stopwatch();
+                watch.Restart();
+                Message message = new Message(new byte[_messageSize]);
+                Task sendTask = _client.SendEventAsync(message);
+                watch.Stop();
+                metrics.SendMs = watch.Elapsed.TotalMilliseconds;
```

Comment:
It's not `sendMs` but `task` was created. We're not able to get `sendMs`  in existing implementation.

Reply:
SendMs measures the time (overheads) it takes to execute the `SendEventAsync` method.
Not sure what you mean by not being able to get sendMs (you mean until we write to the wire?) End-to-end send is actually measured as SendMs + AckMs.